native/acter/src/api/profile.rs

```diff
@@ -19,175 +12,159 @@ use super::{api::FfiBuffer, RUNTIME};
-        let Some(avatar_url) = self.avatar_url.clone() else {
-            bail!("No User Profile found");
-        };
+        let account = self.client.account();
         RUNTIME
             .spawn(async move {
-                let request = MediaRequest {
-                    source: MediaSource::Plain(avatar_url),
-                    format: MediaFormat::File,
-                };
-                let result = client
-                    .media()
-                    .get_media_content(&request, true)
+                let result = account
+                    .get_avatar(MediaFormat::File)
                     .await
-                    .context("Couldn't get media content from user profile")?;
-                Ok(FfiBuffer::new(result))
+                    .context("Couldn't get avatar from account")?;
+                match result {
+                    Some(result) => Ok(FfiBuffer::new(result)),
+                    None => Ok(FfiBuffer::new(vec![])),
+                }
             })
             .await?
     }
 
     pub async fn get_thumbnail(&self, width: u32, height: u32) -> Result<FfiBuffer<u8>> {
-        let client = self.client.clone();
-        let avatar_url = self.avatar_url.clone().unwrap();
+        let account = self.client.account();
         RUNTIME
             .spawn(async move {
                 let size = MediaThumbnailSize {
                     method: Method::Scale,
                     width: UInt::from(width),
                     height: UInt::from(height),
                 };
-                let request = MediaRequest {
-                    source: MediaSource::Plain(avatar_url),
-                    format: MediaFormat::Thumbnail(size),
-                };
-                let result = client
-                    .media()
-                    .get_media_content(&request, true)
+                let result = account
+                    .get_avatar(MediaFormat::Thumbnail(size))
                     .await
-                    .context("Couldn't get media content from user profile")?;
-                Ok(FfiBuffer::new(result))
+                    .context("Couldn't get avatar from account")?;
+                match result {
+                    Some(result) => Ok(FfiBuffer::new(result)),
+                    None => Ok(FfiBuffer::new(vec![])),
+                }
             })
             .await?
     }
 
-    pub fn get_display_name(&self) -> Option<String> {
-        self.display_name.clone()
+    pub async fn get_display_name(&self) -> Result<String> {
+        let account = self.client.account();
+        RUNTIME
+            .spawn(async move {
+                let result = account
+                    .get_display_name()
+                    .await
+                    .context("Couldn't get display name from account")?;
+                match result {
+                    Some(result) => Ok(result),
+                    None => Ok("".to_string()),
+                }
```

Comment:
This is wrong. If the user doesn't have a display name set, the we should return None. An empty string is a valid value and making the assumption that means unset is wrong.

Reply:
`ffi-gen` doesn't support `Result<Option<...>>`.
So I implemented `DispName` crate that has `String? fn text()` method.